test/Garnet.test/RespCustomCommandTests.cs

```diff
@@ -10,13 +10,62 @@
 using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
+using Garnet.common;
 using Garnet.server;
 using NUnit.Framework;
 using NUnit.Framework.Legacy;
 using StackExchange.Redis;
+using Tsavorite.core;
 
 namespace Garnet.test
 {
+    public class LargeGet : CustomProcedure
+    {
+        public override bool Execute(IGarnetApi garnetApi, ArgSlice input, ref MemoryResult<byte> output)
+        {
+            int offset = 0;
+            var key = GetNextArg(input, ref offset);
+            for (int i = 0; i < 120_000; i++)
+            {
+                garnetApi.GET(key, out var outval);
```

Comment:
Will all the returned ArgSlice still be valid after the 120k iterations? seems like if the scratch buffer doubled, the existing ArgSlices are invalidated?

Reply:
The new APIs to record offset and reset the buffer back to any specific offset will help with this.